### PR TITLE
Prevent out of scope log message when not mining

### LINKED ISSUE #219: strange 'mining too far in future' logs seem out of place
#### System information

```
> ./build/bin/geth version
CoreGeth
Version: 1.11.17-unstable
Git Commit: 9dae1199c1cbbe48cb52ac3f170155da6cf9aa92
Git Commit Date: 20201022
Architecture: amd64
Protocol Versions: [65 64 63]
Go Version: go1.15
Operating System: linux
GOPATH=/home/ia/go
GOROOT=/home/ia/go1.15.linux-amd64
```

#### Expected behaviour

Not to be mining too far in the future when not mining.

#### Actual behaviour

```
> eth.mining
false
```

```
Oct 23 11:55:07 ubp52 sh[29718]: INFO [10-23|11:55:07.437] Starting topic registration              topic=LES2@a68ebde7932eccb1
Oct 23 11:55:07 ubp52 sh[29718]: INFO [10-23|11:55:07.440] Mapped network port                      proto=tcp extport=30363 intport=30363 interface=NAT-PMP(192.168.86.1)
Oct 23 11:55:07 ubp52 sh[29718]: INFO [10-23|11:55:07.441] Mapped network port                      proto=udp extport=30363 intport=30363 interface=NAT-PMP(192.168.86.1)
Oct 23 11:55:07 ubp52 sh[29718]: INFO [10-23|11:55:07.443] New local node record                    seq=5 id=844c59be87b8f0ed ip=75.134.144.252 udp=30363 tcp=30363
Oct 23 11:55:08 ubp52 sh[29718]: INFO [10-23|11:55:08.816] Block synchronisation started
Oct 23 11:55:50 ubp52 systemd-journald[552]: Suppressed 3611 messages from core-geth.mordor.service
Oct 23 11:55:50 ubp52 sh[29718]: INFO [10-23|11:55:50.977] Imported new chain segment               blocks=1  txs=0   mgas=0.000 elapsed=9.979ms     mgasps=0.000 number=2541289 hash="5fad38…7adc28" dirty=217.96KiB af=true
Oct 23 11:55:54 ubp52 sh[29718]: INFO [10-23|11:55:54.512] Imported new chain segment               blocks=1  txs=0   mgas=0.000 elapsed=9.802ms     mgasps=0.000 number=2541290 hash="39e6ad…676b96" dirty=221.07KiB af=true
Oct 23 11:55:54 ubp52 sh[29718]: INFO [10-23|11:55:54.513] Mining too far in the future             wait=4s
Oct 23 11:56:08 ubp52 sh[29718]: INFO [10-23|11:56:08.388] Deep froze chain segment                 blocks=11867 elapsed=1.760s      number=2451290 hash="94fe80…1840c8"
Oct 23 11:56:11 ubp52 sh[29718]: INFO [10-23|11:56:11.698] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=5.088ms     mgasps=0.000 number=2541291 hash="4f1485…841027" dirty=224.44KiB af=true
Oct 23 11:56:23 ubp52 sh[29718]: INFO [10-23|11:56:23.640] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=10.127ms    mgasps=0.000 number=2541292 hash="697ccd…fd6a8e" dirty=227.55KiB af=true
Oct 23 11:56:28 ubp52 sh[29718]: INFO [10-23|11:56:28.592] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=9.491ms     mgasps=0.000 number=2541293 hash="f27fd2…6658c5" dirty=230.93KiB af=true
Oct 23 11:57:03 ubp52 sh[29718]: INFO [10-23|11:57:03.017] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=3.083ms     mgasps=0.000 number=2541294 hash="c59b48…e71c06" dirty=234.31KiB af=true
Oct 23 11:57:04 ubp52 sh[29718]: INFO [10-23|11:57:04.002] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=9.428ms     mgasps=0.000 number=2541295 hash="456450…e59c4f" dirty=237.42KiB af=true
Oct 23 11:57:04 ubp52 sh[29718]: INFO [10-23|11:57:04.003] Mining too far in the future             wait=6s
Oct 23 11:57:08 ubp52 sh[29718]: INFO [10-23|11:57:08.509] Deep froze chain segment                 blocks=5     elapsed=120.976ms   number=2451295 hash="7807b7…0c871e"
Oct 23 11:57:15 ubp52 sh[29718]: INFO [10-23|11:57:15.283] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=4.246ms     mgasps=0.000 number=2541296 hash="353420…80dd42" dirty=240.80KiB af=true
Oct 23 11:57:59 ubp52 sh[29718]: INFO [10-23|11:57:59.115] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=9.765ms     mgasps=0.000 number=2541297 hash="9da838…df6838" dirty=243.91KiB af=true
Oct 23 11:58:01 ubp52 sh[29718]: INFO [10-23|11:58:01.381] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=9.887ms     mgasps=0.000 number=2541298 hash="78a810…a0ce5f" dirty=247.02KiB af=true
Oct 23 11:58:01 ubp52 sh[29718]: INFO [10-23|11:58:01.381] Mining too far in the future             wait=5s
Oct 23 11:58:08 ubp52 sh[29718]: INFO [10-23|11:58:08.626] Deep froze chain segment                 blocks=3     elapsed=116.439ms   number=2451298 hash="e883cc…09b51c"
Oct 23 11:58:08 ubp52 sh[29718]: INFO [10-23|11:58:08.906] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=3.957ms     mgasps=0.000 number=2541299 hash="8b03cf…66a92d" dirty=250.40KiB af=true
Oct 23 11:58:14 ubp52 sh[29718]: INFO [10-23|11:58:14.880] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=10.391ms    mgasps=0.000 number=2541300 hash="652a30…38b045" dirty=253.78KiB af=true
Oct 23 11:58:22 ubp52 sh[29718]: INFO [10-23|11:58:22.997] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=10.544ms    mgasps=0.000 number=2541301 hash="2d7b20…9d5a63" dirty=256.89KiB af=true
Oct 23 11:58:33 ubp52 sh[29718]: INFO [10-23|11:58:33.202] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=3.296ms     mgasps=0.000 number=2541302 hash="7344e4…4dbe36" dirty=260.00KiB af=true
Oct 23 11:58:51 ubp52 sh[29718]: INFO [10-23|11:58:51.499] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=9.344ms     mgasps=0.000 number=2541303 hash="83a2cb…6f624e" dirty=263.11KiB af=true
Oct 23 11:58:51 ubp52 sh[29718]: INFO [10-23|11:58:51.676] Imported new chain segment               blocks=1     txs=0   mgas=0.000 elapsed=10.024ms    mgasps=0.000 number=2541304 hash="43ca9c…f71129" dirty=266.22KiB af=true
Oct 23 11:58:51 ubp52 sh[29718]: INFO [10-23|11:58:51.676] Mining too far in the future             wait=7s

```

#### Steps to reproduce the behaviour

```
./build/bin/geth --mordor --http --http.port 8545 --http.corsdomain=\'*\' --port 30303 --verbosity 5 --datadir /home/ia/SAMSUNG_T5/e/clients/core-geth/mordor --ethash.dagdir /home/ia/SAMSUNG_T5/e/clients/core-geth/mordor/.ethash --miner.gaslimit=100000000 --miner.gastarget=64000000 --ipcpath /tmp/core-geth_mordor.ipc --lightserv 10 --keystore /home/ia/SAMSUNG_T5/e/keys/mordor
```

### PR DESCRIPTION
**Pull request by ziogaschr**
Resolves #219.

There is no reason to print `mining too far in future` in logs when not mining.

This is being triggered from the following flow:

1. https://github.com/etclabscore/core-geth/blob/5384bc91bc8d11e8b5878c4d9fa09b1969426fc8/miner/worker.go#L373-L376
2. https://github.com/etclabscore/core-geth/blob/5384bc91bc8d11e8b5878c4d9fa09b1969426fc8/miner/worker.go#L345-L354
3. https://github.com/etclabscore/core-geth/blob/5384bc91bc8d11e8b5878c4d9fa09b1969426fc8/miner/worker.go#L434-L435
4. https://github.com/etclabscore/core-geth/blob/5384bc91bc8d11e8b5878c4d9fa09b1969426fc8/miner/worker.go#L861-L868
5. https://github.com/etclabscore/core-geth/blob/5384bc91bc8d11e8b5878c4d9fa09b1969426fc8/miner/worker.go#L986-L996

So basically, until step 4 it generates the newWork, while in step 5 where it actually commits it, it won't if not mining. 
For this reason, I suggest that we remove the misleading log while still in 4.

### CODE REVIEW

**Review by meowsbits — COMMENTED** [review]
I'm generally on the other side of removing or silencing _any_ logs... :joy: especially when actual causal behavior isn't modified. 

What do you think about modifying the log level conditionally instead? Like

```go
logFn := log.Trace
if w.isRunning() {
    logFn := log.Info
}
logFn("Mining too far...")
```

**Comment by ziogaschr** [comment]
Nice idea man 👍 . The word `Mining` triggered me in the mining path only.

**Review by meowsbits — APPROVED** [review]

**Review by iquidus — APPROVED** [review]
